chain/client/tests/catching_up.rs

```diff
@@ -641,4 +642,209 @@ mod tests {
         })
         .unwrap();
     }
+
+    enum OrphansPhases {
+        WaitingForSendingTx,
+        HoldingBlock,
+        ValidatingOrphans,
+    }
+
+    #[test]
+    fn test_catchup_orphans_under_epoch() {
+        test_catchup_orphans_common(2, 5)
+    }
+
+    #[test]
+    fn test_catchup_orphans_over_epoch() {
+        test_catchup_orphans_common(2, 10)
+    }
+
+    fn test_catchup_orphans_common(height_hold: u64, height_release: u64) {
+        if !cfg!(feature = "expensive_tests") {
+            return;
+        }
+        assert!(height_hold > 1);
+        let validator_groups = 1;
+        init_integration_logger();
+        System::run(move || {
+            let connectors: Arc<RwLock<Vec<(Addr<ClientActor>, Addr<ViewClientActor>)>>> =
+                Arc::new(RwLock::new(vec![]));
+
+            let (validators, key_pairs) = get_validators_and_key_pairs();
+
+            let phase = Arc::new(RwLock::new(OrphansPhases::WaitingForSendingTx));
+            let chunk_hashes = Arc::new(RwLock::new(HashMap::new()));
+            let chunk_hashes1 = chunk_hashes.clone();
+            let chunk_heights = Arc::new(RwLock::new(HashSet::<BlockIndex>::new()));
+            let block_heights = Arc::new(RwLock::new(HashSet::<BlockIndex>::new()));
+            let connectors_done = Arc::new(RwLock::new(HashSet::new()));
+
+            let add_hash = move |hash: ChunkHash, height| match chunk_hashes1
+                .write()
+                .unwrap()
+                .entry(hash.clone())
```

Comment:
`entry(...).or_insert_with(...)`

Reply:
I can't get what are you talking about. It's should be closure because of compiler requirements.